shared/src/commonMain/kotlin/com/devil/phoenixproject/domain/usecase/RoutineTimeEstimator.kt

```
@@ -2,65 +2,277 @@ package com.devil.phoenixproject.domain.usecase
-     * Estimate routine duration in seconds.
-     * Uses historical average set duration per exercise when available,
-     * falls back to calculation from rest times + estimated 45s per set.
+     * Estimate routine duration.
+     *
+     * @param routine The routine to estimate
+     * @param profileId The user profile ID (for querying historical data)
+     * @return Estimated duration with range bounds for AMRAP routines
      */
-    suspend fun estimateRoutineDuration(routine: Routine): RoutineTimeEstimate {
-        var totalHistoricalMs = 0L
+    suspend fun estimateRoutineDuration(routine: Routine, profileId: String): RoutineTimeEstimate {
```

detekt (1.23.8)

[warning] The function estimateRoutineDuration appears to be too complex based on Cognitive Complexity (complexity: 34). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)